src/PointwiseFunctions/AnalyticSolutions/Xcts/Schwarzschild.hpp

```diff
@@ -0,0 +1,234 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <ostream>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/Systems/Xcts/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Xcts::Solutions {
+
+/// Various coordinate systems in which to express the Schwarzschild solution
+enum class SchwarzschildCoordinates {
+  /*!
+   * \brief Isotropic Schwarzschild coordinates
+   *
+   * These arise from the canonical Schwarzschild coordinates by the radial
+   * transformation
+   *
+   * \f{equation}
+   * r = \bar{r}\left(1+\frac{M}{2\bar{r}}\right)^2
+   * \f}
+   *
+   * (Eq. (1.61) in \cite BaumgarteShapiro) where \f$r\f$ is the canonical
+   * Schwarzschild radius, also referred to as "areal" radius because it is
+   * defined such that spheres with constant \f$r\f$ have the area \f$4\pi
+   * r^2\f$, and \f$\bar{r}\f$ is the "isotropic" radius. In the isotropic
+   * radius the Schwarzschild spatial metric is conformally flat
+   *
+   * \f{equation}
+   * \gamma_{ij}=\psi^4\eta_{ij} \quad \text{with conformal factor} \quad
+   * \psi=\left(1+\frac{M}{2\bar{r}}\right)^4
+   * \f}
+   *
+   * (Table 2.1 in \cite BaumgarteShapiro). Its lapse transforms to
+   *
+   * \f{equation}
+   * \alpha=\frac{1-M/(2\bar{r})}{1+M/(2\bar{r})}
+   * \f}
+   *
+   * and the shift vanishes \f$\beta^i=0\f$ as it does in areal Schwarzschild
+   * coordinates. The solution also remains maximally sliced \f$K=0\f$.
+   *
+   * The Schwarzschild horizon in these coordinates is at
+   * \f$\bar{r}=\frac{M}{2}\f$ due to the radial transformation from \f$r=2M\f$.
+   */
+  Isotropic,
+};
+
+std::ostream& operator<<(std::ostream& os,
+                         const SchwarzschildCoordinates& coords) noexcept;
+
+/*!
+ * \brief Schwarzschild spacetime in general relativity
+ *
+ * This class implements the Schwarzschild solution with mass parameter
+ * \f$M=1\f$ in various coordinate systems. See the entries of the
```

Comment:
Why is total mass not an option?

Reply:
We'll only need this solution to test the XCTS system, boundary conditions, the XCTS executable and so on, but probably not for anything that involves a binary (we'll have the Kerr solution for that, in Kerr-Schild and Harmonic coords), so the mass parameter would only be an unnecessary global rescaling.

Reply:
At least for the evolution code it is extremely convenient to be able to test the (local) domain structure, etc. with BHs that are the same as those used in a binary. For example, for high-mass ratios different constraint damping is necessary and so being able to test on a single BH was useful there. The other reason I think it's good to add now is that it's much easier for you to do it now than someone else in the future